pkg/engine/engine.go

```diff
@@ -694,18 +694,18 @@ func (e *Engine) verificationOverlapWorker(ctx context.Context) {
 					}, res)
 
 					// Remove the detector from the list of detectors with results.
-					delete(detectorsWithResult, detector)
+					delete(detectorKeysWithResults, detector.Key)
 				}
 				chunkSecrets[key] = struct{}{}
 			}
 		}
 
-		for detector := range detectorsWithResult {
+		for key := range detectorKeysWithResults {
 			wgDetect.Add(1)
 			chunk.chunk.Verify = e.verify
 			e.detectableChunksChan <- detectableChunk{
 				chunk:    chunk.chunk,
-				detector: detector,
+				detector: e.ahoCorasickCore.GetDetectorByKey(key),
```

Comment:
pretty sure this is not gonna happen but might be worth it to check for nil before sending the chunk since `GetDetectorByKey` could return nil